File: Task2/task2.go
package main
import (
	"fmt"
)
func main() {
	var numbers int
	fmt.Scan(&numbers)
	if numbers <= 0 {
		fmt.Println("Быть не может")
		return
	}
	clumba := make([]int, 0)
	max := 0
	for i := 0; i < numbers; i++ {
		var flower int
		fmt.Scan(&flower)
		if flower > 0 {
			clumba = append(clumba, flower)
		}
		if flower > max {
			max = flower
		}
	}
	var poliv int
	fmt.Scan(&poliv)
	if poliv < max {
		fmt.Println("Как купишь новую лейку - приходи.")
		return
	}
	step := 0
	perem := poliv
	for i := 0; i < len(clumba); i++ {
		if perem < clumba[i] {
			step += i*2 + 1
			perem = poliv
		} else {
			step += 1
		}
		perem -= clumba[i]
	}
	fmt.Println(step)
}

File: Task2/task2.java
import java.util.ArrayList;
import java.util.List;
import java.util.Scanner;
class Main {
    public static void main(String[] args) {
        Scanner in = new Scanner(System.in);
        int numbers = in.nextInt();
        int max = 0;
        if (numbers <= 0) {
            System.out.println("Быть не может");
            return; 
        }
        List<Integer> clumba = new ArrayList<>();
        for (int i = 0; i < numbers; ++i) {
            int flower = in.nextInt();
            if (flower > 0) {
                clumba.add(flower);
            }
            if (flower > max) {
                max = flower;
            }
        }
        int poliv = in.nextInt();
        if (poliv < max) {
            System.out.println("Как купишь новую лейку - приходи.");
            return;
        }
        int step = 0;
        int perem = poliv;
        for (int i = 0; i < clumba.size(); i++) {
            if (perem < clumba.get(i)) {
                step = step + (i * 2 + 1);
                perem = poliv;
            } else {
                step += 1;
            }
            perem -= clumba.get(i);
        }
        System.out.println(step);
    }
}
